governance/tsc/tsc-2024-04-29.md

```diff
@@ -0,0 +1,238 @@
+
+martin: our proposal is:
+
+martin: we think these tools are knowingly trained on copyrighted work, so their output is a liability for the project
+
+martin: these tools make it too easy to generate large amounts of plausible but dangerously incorrect contributions, which we do not have the capacity for as maintainers
+
+martin: we've based this ideas in the discussion in the [Gentoo mailing list](https://www.mail-archive.com/gentoo-dev@lists.gentoo.org/msg99042.html)
+
+martin: our proposal is to disallow the use of generative AI tools when contributing to Servo projects, including code, documentation, and bug reports
+
+nico: I don't personally use such tools, but I know they are popular. The argument that it would be more serious contributors off. It's clear that we don't want low effort entirely generated with AI. I think senior software engineers are starting to use these tools. I wouldn't oppose this policy, but...
+
+gregory: I tried out GitHub Copilot if you are contributing to open source like Servo and it looks pretty well. I wrote a little Rust script a few days ago and it guesses what you're going to do, but it does make a lot of suggestions. If people are working with that and accepting suggestions and just opening a PR without having a look at it themselves. Even I was caught-on with something. I wanted to refactor something -- it uses an API that inserted and moved and only realized it later. What I'm trying to say is we could decide to ban it, but on the other hand there are also some opportunities -- for instance to generate bindings. On the DOM side of things it could be nice if it could generate some more boilerplate with AI.
+
+gregory: We have patterns in the code, but GitHub Copilot is not able to recognize them. In the short term if it's experienced as spam, then you could have a temporary ban. With Stack Overflow they have done that.
+
+rego: People can always use these tools and you may not notice that. If you have a nice editor that autocompletes, that's not a problem. It's these kind of PRs. If people are fine, maybe we can draft this proposal.
+
+### Crowdfunding
+
+rego: We still have the LFX platform open. We are still talking to the LF to transfer the money to open collective. That hasn't happened yet, but OC and GitHub sponsors have a bunch now. In a month or so, 2500 collected. We can talk about how to use this money in the future. Just an update on the situation.
+
+### Outreach
+
+rego: There is a GOSIM event happening in the Netherland in one week. Gregory will be talking about Servo there. Nico & Jonathan will Servo there.
+gregory: It's next week.
+
+rego: https://europe2024.gosim.org/
+
+rego: The next day is RustNL. Rakhi is also attending that one. There will be a bunch of Servo people around and Rust people.
+
+rego: https://2024.rustnl.org/
+
+rego: In a month there is the Web Engines Hackfest in June. Registration is still open. Nico and the Igalia team will be there. Registration is still open.
```

Comment:
nit: last sentence is duplicate, we can remove it

Reply:
```suggestion
rego: In a month there is the Web Engines Hackfest in June. Nico and the Igalia team will be there. Registration is still open.
```